src/protocol/control_layer/ControlMessageFactory.js

```diff
@@ -1,68 +1,33 @@
-import UnicastMessageFactory from './unicast_message/UnicastMessageFactory'
-import SubscribeResponse from './subscribe_response/SubscribeResponse'
-import SubscribeResponseFactory from './subscribe_response/SubscribeResponseFactory'
-import UnsubscribeResponse from './unsubscribe_response/UnsubscribeResponse'
-import UnsubscribeResponseFactory from './unsubscribe_response/UnsubscribeResponseFactory'
-import ResendResponseResending from './resend_response_resending/ResendResponseResending'
-import ResendResponseResendingFactory from './resend_response_resending/ResendResponseResendingFactory'
-import ResendResponseResent from './resend_response_resent/ResendResponseResent'
-import ResendResponseResentFactory from './resend_response_resent/ResendResponseResentFactory'
-import ResendResponseNoResend from './resend_response_no_resend/ResendResponseNoResend'
-import ResendResponseNoResendFactory from './resend_response_no_resend/ResendResponseNoResendFactory'
-import ResendLastRequestV1 from './resend_request/ResendLastRequestV1'
-import ResendLastRequestFactory from './resend_request/ResendLastRequestFactory'
-import ResendFromRequestV1 from './resend_request/ResendFromRequestV1'
-import ResendFromRequestFactory from './resend_request/ResendFromRequestFactory'
-import ResendRangeRequestV1 from './resend_request/ResendRangeRequestV1'
-import ResendRangeRequestFactory from './resend_request/ResendRangeRequestFactory'
-import ErrorResponse from './error_response/ErrorResponse'
-import ErrorResponseFactory from './error_response/ErrorResponseFactory'
-import ControlMessageV0Factory from './ControlMessageV0Factory'
 
 const factoryByMessageType = {}
-factoryByMessageType[PublishRequest.TYPE] = PublishRequestFactory
-factoryByMessageType[SubscribeRequest.TYPE] = SubscribeRequestFactory
-factoryByMessageType[UnsubscribeRequest.TYPE] = UnsubscribeRequestFactory
-factoryByMessageType[BroadcastMessage.TYPE] = BroadcastMessageFactory
-factoryByMessageType[UnicastMessage.TYPE] = UnicastMessageFactory
-factoryByMessageType[SubscribeResponse.TYPE] = SubscribeResponseFactory
-factoryByMessageType[UnsubscribeResponse.TYPE] = UnsubscribeResponseFactory
-factoryByMessageType[ResendResponseResending.TYPE] = ResendResponseResendingFactory
-factoryByMessageType[ResendResponseResent.TYPE] = ResendResponseResentFactory
-factoryByMessageType[ResendResponseNoResend.TYPE] = ResendResponseNoResendFactory
-factoryByMessageType[ResendLastRequestV1.TYPE] = ResendLastRequestFactory
-factoryByMessageType[ResendFromRequestV1.TYPE] = ResendFromRequestFactory
-factoryByMessageType[ResendRangeRequestV1.TYPE] = ResendRangeRequestFactory
-factoryByMessageType[ErrorResponse.TYPE] = ErrorResponseFactory
-
-export default class ControlMessageFactory {
-    static buildControlMessage(version, type, typeSpecificArgsArray) {
-        return factoryByMessageType[type].deserialize(version, typeSpecificArgsArray)
-    }
 
+class ControlMessageFactory {
     static deserialize(msg) {
```

Comment:
- Please move to `ControlMessage.deserialize(msg)`
- Note the comment in `BroadcastMessage` about generalizing the `deserialize` method. Here, the line 
`factoryByMessageType[messageType].deserialize(messageVersion, args)`
could be replaced by the 1-2 lines in the "generalized" `deserialize` suggestion there
- => Will there be need for `registerFactory` anymore?